baseplate/clients/requests.py

```diff
@@ -200,9 +205,22 @@ def _add_span_context(self, span: Span, request: PreparedRequest) -> None:
 
     def send(self, request: PreparedRequest, **kwargs: Any) -> Response:
         """Send a :py:class:`~requests.PreparedRequest`."""
-        with self.span.make_child(f"{self.name}.request") as span:
+        # we cannot use a context manager anymore to create the span here as we need all the tags
+        # to be set properly before the span can start for prometheus to pick them up
+        # this is why we need to call `self.span.make_child(...)` manually and subsequently:
+        # - span.start()
+        # - span.finish()
```

Comment:
I wonder if it'd be worth adding something along these lines so that we could leverage the context manager...

```diff
diff --git a/baseplate/__init__.py b/baseplate/__init__.py
index 54e7492..4e0d0e3 100644
--- a/baseplate/__init__.py
+++ b/baseplate/__init__.py
@@ -592,6 +592,11 @@ class Span:
         for observer in self.observers:
             observer.on_start()
 
+    def with_tags(self, tags: Dict[str,Any]):
+        for k, v in tags:
+            self.set_tag(k, v)
+        return self
+
     def set_tag(self, key: str, value: Any) -> None:
         """Set a tag on the span.
 
```

Reply:
Yeah it's similar to the builder pattern. I'm just unfamiliar enough with context managers that I didn't think to try and do it.
It seem to be working just fine though.